### ISSUE TITLE
GixSQL 1.0.9 compiler warnings - and fix proposal

### ISSUE THREAD

**Issue by GitMensch** [mridoni/gix#53]
~~~
DataSourceInfo.cpp: In member function 'std::__cxx11::string DataSourceInfo::dump(bool)':
DataSourceInfo.cpp:234:1: warning: no return statement in function returning non-void [-Wreturn-type]

SqlVar.cpp:31: warning: "SIGN_LENGTH" redefined
 #define SIGN_LENGTH ((int)1)

SqlVar.cpp:32: warning: "TERMINAL_LENGTH" redefined
 #define TERMINAL_LENGTH ((int)1)
 ~~~
 
The first one should likely be fixed by:
 ```diff
 	return s;
+#else
+	return "";
 #endif
}
```

The later two are easy to fix: just remove those either in SqlVar.cpp or in utils.h.

**Comment by GitMensch** [comment]
Second thought: it would be better for the first issue to move the `#ifdef DEBUG` guard around the method (also in the definition) to ensure that it is only called when compiled for debug, too (otherwise the "no-op calls" still use cpu-cycles).

**Comment by GitMensch** [comment]
Rechecked: both warnings still occur with 1.0.10

**Comment by mridoni** [comment]
Yes, I could easily have fixed this, but obviously escaped triage...

**Comment by GitMensch** [comment]
escaped again with 1.10.11

**Comment by GitMensch** [comment]
Actually no - the first part seems to be fixed, the second part still open and new:

```
GixPreProcessor.cpp:30: warning: "SET_ERR" redefined
 #define SET_ERR(I,S) err_data.err_code = I; err_data.err_messages.push_back(S)

In file included from GixPreProcessor.h:28,
                 from GixPreProcessor.cpp:21:
ITransformationStep.h:25: note: this is the location of the previous definition
 #define SET_ERR(I,S) owner->err_data.err_code = I; owner->err_data.err_messages.push_back(S)



DbInterfacePGSQL.cpp:29: warning: "LOG_DEBUG" redefined
 #define LOG_DEBUG(file, func, format, ...) logger->log_debug(file, func, format, ##__VA_ARGS__)

In file included from DbInterfacePGSQL.cpp:25:
../libgixsql/Logger.h:48: note: this is the location of the previous definition
 #define LOG_DEBUG(file, func, format, ...) logger.log_debug(file, func, format, ##__VA_ARGS__)

DbInterfacePGSQL.cpp:30: warning: "LOG_ERROR" redefined
 #define LOG_ERROR(format, ...) logger->log_error(format, ##__VA_ARGS__)

In file included from DbInterfacePGSQL.cpp:25:
../libgixsql/Logger.h:49: note: this is the location of the previous definition
 #define LOG_ERROR(format, ...) logger.log_error(format, ##__VA_ARGS__)
```

**Comment by mridoni** [comment]
Fixed in my internal repository, will be in v1.0.12 (and in v1.0.11b)

**Comment by GitMensch** [comment]
You may want to double-check if the warnings are both fixed with/without defining `DEBUG`.

**Comment by mridoni** [comment]
> You may want to double-check if the warnings are both fixed with/without defining `DEBUG`.

I'll check the `LOG_*` macros, for the other cases it is very unlikely: for the first case `linq_iterators.hpp` the `return` was actually missing. For the second case the macro was redefined (with a different definition) so I just renamed it.

**Comment by mridoni** [comment]
> I'll check the `LOG_*` macros

Fixed in the internal repository, ready for v1.0.12

**Comment by GitMensch** [comment]
Released, so should be fixed.
